dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -192,6 +194,18 @@ public T editHosted(@ApiParam(hidden = true) @Auth User user,
         return saveVersion(user, entryId, entry, version, versionSourceFiles, Optional.empty());
     }
 
+    /**
+     * Checks that all sourcefiles have absolute paths
+     * @param sourceFiles
+     */
+    protected void checkAbsolutePath(Set<SourceFile> sourceFiles) {
+        sourceFiles.forEach(((SourceFile sourceFile) -> {
+            if (!(sourceFile.getPath().startsWith("/"))) {
+                throw new CustomWebApplicationException("Hosted files must all be absolute path, but " + sourceFile.getPath() + " is not.", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Sounds odd; maybe "Hosted files must have an absolute path", although that sounds odd too.